javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,385 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+class LinkedList[E]() extends AbstractSequentialList[E]
+    with List[E] with Deque[E] with Cloneable with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  import LinkedList._
+
+  var head: Node[E] = null
+  var last: Node[E] = null
+
+  var _size = 0
+
+  def getFirst(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekFirst()
+
+  def getLast(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekLast()
+
+  def removeFirst(): E = {
+    if (isEmpty()) throw new NoSuchElementException()
+    val ret = head.value
+
+    val oldHead = head
+    head = oldHead.next
+
+    if (head ne null) {
+      head.prev = null
+
+      if (head eq last) {
+        head.next = null
+        last = null
+      }
```

Comment:
This if is neither necessary nor correct. If `head eq last`, it means that *after* the operation, `head` is the sole element of the list, and it also the last one (and it was already the last one). So `head.next` should be `null`, but it is already the case, and `last` should be `head`, but it is already the case. So these four lines can entirely go away.